pkg/yurthub/cachemanager/storage_wrapper_test.go

```diff
@@ -0,0 +1,154 @@
+			Kind:       "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "mypod1",
+			Namespace:       "default",
+			ResourceVersion: "1",
+		},
+	})
+
+	defer clearDir(dir)
+
+	dStorage, err := disk.NewDiskStorage(dir)
+	if err != nil {
+		t.Errorf("failed to create disk storage, %v", err)
+	}
+	sWrapper := NewStorageWrapper(dStorage)
+
+	t.Run("Test create storage", func(t *testing.T) {
+		err = sWrapper.Create("kubelet/pods/default/mypod1", pod)
+		if err != nil {
+			t.Errorf("failed to create obj, %v", err)
+		}
+		obj, err := sWrapper.Get("kubelet/pods/default/mypod1")
+		if err != nil {
+			t.Errorf("failed to create obj, %v", err)
+		}
+		accessor := meta.NewAccessor()
+		name, _ := accessor.Name(obj)
+		if name != "mypod1" {
+			t.Errorf("the name is not expected, expect mypod1, get %s", name)
+		}
+	})
+
+	t.Run("Test update storage", func(t *testing.T) {
+		updatePod := runtime.Object(&v1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Pod",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            "mypod1",
+				Namespace:       "default",
+				ResourceVersion: "1",
+				Labels: map[string]string{
+					"tag": "test",
+				},
+			},
+		})
+		err = sWrapper.Update("kubelet/pods/default/mypod1", updatePod)
+		if err != nil {
+			t.Errorf("failed to update obj, %v", err)
+		}
+		obj, err := sWrapper.Get("kubelet/pods/default/mypod1")
+		if err != nil {
+			t.Errorf("unexpected error occurs, %v", err)
+		}
+		accessor := meta.NewAccessor()
+		labels, _ := accessor.Labels(obj)
+		if vaule, ok := labels["tag"]; ok {
+			if vaule != "test" {
+				t.Errorf("failed to get label, expect test, get %s", vaule)
+			}
+		} else {
+			t.Errorf("unexpected error occurs, the label `tag` is not existed")
+		}
+
+	})
+
+	t.Run("Test list obj and keys", func(t *testing.T) {
+		_, err = sWrapper.ListKeys("kubelet/pods/default")
+		if err != nil {
+			t.Errorf("failed to list keys, %v", err)
+		}
+		_, err = sWrapper.List("kubelet/pods/default")
+		if err != nil {
+			t.Errorf("failed to list objs, %v", err)
+		}
+	})
+
+	t.Run("Test replace obj", func(t *testing.T) {
+		err = sWrapper.Replace("kubelet/pods/default", map[string]runtime.Object{
+			"kubelet/pods/default/mypod1": runtime.Object(&v1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "Pod",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "mypod1",
+					Namespace:       "default",
+					ResourceVersion: "1",
+					Labels: map[string]string{
+						"tag": "test",
+					},
+				},
+			}),
+		})
+		if err != nil {
+			t.Errorf("failed to replace objs, %v", err)
+		}
+	})
+
+	t.Run("Test delete storage", func(t *testing.T) {
+		err = sWrapper.Delete("kubelet/pods/default/mypod1")
+		if err != nil {
+			t.Errorf("failed to delete obj, %v", err)
+		}
+		_, err = sWrapper.Get("kubelet/pods/default/mypod1")
+		if err != storage.ErrStorageNotFound {
+			t.Errorf("unexpected error occurs, %v", err)
+		}
```

Comment:
Here may need to verify the existence of the object in the cache for the same reason as above.